src/btree/bt_sync.c

```diff
@@ -317,14 +261,28 @@ __sync_ref_obsolete_check(WT_SESSION_IMPL *session, WT_REF *ref, WT_REF_LIST *rl
         newest_stop_ts = addr.ta.newest_stop_ts;
         newest_stop_durable_ts =
           addr.ta.newest_stop_ts == WT_TS_MAX ? WT_TS_MAX : addr.ta.newest_stop_durable_ts;
+        if (addr.type == WT_ADDR_LEAF)
+            ovfl_items = true;
         obsolete = __wt_txn_visible_all(session, newest_stop_txn, newest_stop_durable_ts);
     } else
         tag = "unexpected page state";
 
-    /* If the page is obsolete, add it to the list of pages to be evicted. */
+    /*
+     * If the page is obsolete, set the page read generation number to a lower value to evict it
+     * sooner.
+     */
     if (obsolete) {
-        WT_ERR(__sync_ref_list_add(session, rlp, ref));
-        hazard = false;
+        /*
+         * Dirty the obsolete page with overflow items to let the page reconciliation remove all the
+         * overflow items.
+         */
+        if (ovfl_items) {
+            WT_ERR(__wt_page_modify_init(session, ref->page));
+            __wt_page_modify_set(session, ref->page);
+        }
+
```

Comment:
May be add a comment here.

Reply:
Maybe just move the comment at line 271 to line 284?

Reply:
Fixed.